pkg/report/sarif_test.go

```diff
@@ -0,0 +1,178 @@
+package report_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	dbTypes "github.com/aquasecurity/trivy-db/pkg/types"
+	"github.com/aquasecurity/trivy/pkg/report"
+	"github.com/aquasecurity/trivy/pkg/types"
+)
+
+type sarifTemplate struct {
+	Schema string      `json:"$schema"`
+	Runs   []sarifRuns `json:"runs"`
+}
```

Comment:
Any reason we cannot use this struct?
https://github.com/owenrumney/go-sarif/blob/b94bb5ae9ee751a86f9ef2cbc44b1113bc2993de/v2/sarif/sarif.go#L21-L28

Reply:
I wanted to create test data with required fields, but `omitempty` allows to skip (forget) it in tests. 
also some of required fields maybe don't exist in `go-sarif`. we'd reveal it.
but if this isn't important, I'll use specified structure

Reply:
>I wanted to create test data with required fields, but omitempty allows to skip (forget) it in tests.

I think it is OK if we are careful about expected data.

>also some of required fields maybe don't exist in go-sarif

Do you mean we cannot use `go-sarif` to generate the SARIF report if it is missing some required fields?

Reply:
I mean that we can detect missing fields and ask @owenrumney to add them. 
ok, i'll use `go-sarif`'s structs for tests.

Reply:
>I mean that we can detect missing fields and ask @owenrumney to add them.

I understand. I think it should be done aside from unit tests.